thoth/storages/graph/postgres.py

```diff
@@ -1733,6 +1734,66 @@ def get_si_unanalyzed_python_package_versions_count_all(
 
             return query.count()
 
+    def get_si_unanalyzed_not_downloadable_python_package_versions_all(
+        self,
+        start_offset: int = 0,
+        count: Optional[int] = DEFAULT_COUNT,
+        distinct: bool = True,
+        randomize: bool = True,
+    ) -> List[Tuple[str, str, str]]:
+        """Retrieve solved Python package versions in Thoth Database, that are not anaylyzed by SI and not downloadable. 
+        Examples:
+        >>> from thoth.storages import GraphDatabase
+        >>> graph = GraphDatabase()
+        >>> graph.get_si_unanalyzed_not_downloadable_python_package_versions_all()
+        [('crossbar', '0.10.0', 'https://pypi.org/simple'), ('tensorflow', '1.11.0', 'https://pypi.org/simple')]
+        """
+        with self._session_scope() as session:
+            query = self._construct_si_unanalyzed_python_package_versions_query(
+                session,
+                is_downloadable=False
+            )
+
+            query = query.join(PythonPackageIndex).with_entities(
+                PythonPackageVersion.package_name,
+                PythonPackageVersion.package_version,
+                PythonPackageIndex.url,
+            )
+
+            if randomize:
+                query = query.order_by(func.random())
+
+            query = query.offset(start_offset).limit(count)
+
+            if distinct:
+                query = query.distinct()
+
+            return query.all()
+
+    def get_si_unanalyzed_not_downloadable_python_package_versions_count_all(
+        self,
+        index_url: Optional[str] = None,
+        *,
+        distinct: bool = False,
+    ) -> int:
+        """Get SI unanalyzed not downloadable Python package versions number in Thoth Database."""
+        with self._session_scope() as session:
+            query = self._construct_si_unanalyzed_python_package_versions_query(
+                session,
+                is_downloadable=False
```

Comment:
What about having this as a parameter in both cases?

Reply:
you mean without any default in `_construct_si_unanalyzed_python_package_versions_query`?

Reply:
Not to hardcode "not_downloadable" into method name `get_si_unanalyzed_not_downloadable_python_package_versions_all`. Have `is_downloadable` as a parameter (as in case of `is_missing` in other methods).

Reply:
Update in 1fa6f10b40385757ee64f7d11c36999b8b2d4646